lib/samples/navigate_route_with_rerouting/navigate_route_with_rerouting.dart

```diff
@@ -0,0 +1,532 @@
+  late Graphic _remainingRouteGraphic;
+  late Graphic _routeTraveledGraphic;
+  // A PolylineBuilder to store the traversed route.
+  var _traversedRouteBuilder =
+      PolylineBuilder(spatialReference: SpatialReference.wgs84);
+  // San Diego Convention Center.
+  final _startPoint = ArcGISPoint(
+    x: -117.160386727,
+    y: 32.706608,
+    spatialReference: SpatialReference.wgs84,
+  );
+  // RH Fleet Aerospace Museum.
+  final _endPoint = ArcGISPoint(
+    x: -117.146679,
+    y: 32.730351,
+    spatialReference: SpatialReference.wgs84,
+  );
+  // Indicate whether the route is being navigated.
+  var _isNavigating = false;
+  var _needRecenter = false;
+  // Variables to show the remaining distance, time, and next direction.
+  var _routeStatus = '';
+  var _remainingDistance = '';
+  var _remainingTime = '';
+  var _nextDirection = '';
+
+  @override
+  void initState() {
+    // Initialize the text-to-speech plugin.
+    _flutterTts = FlutterTts();
+    _flutterTts.setSpeechRate(0.5);
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    Container(
+                      decoration: BoxDecoration(
+                        shape: BoxShape.circle,
+                        color: Theme.of(context).primaryColorLight,
+                      ),
+                      child: IconButton(
+                        onPressed: _isNavigating ? stop : null,
+                        color: Colors.white,
+                        icon: const Icon(Icons.stop),
+                      ),
+                    ),
+                    Container(
+                      decoration: BoxDecoration(
+                        shape: BoxShape.circle,
+                        color: Theme.of(context).primaryColorLight,
+                      ),
+                      child: IconButton(
+                        onPressed: _isNavigating ? null : start,
+                        color: Colors.white,
+                        icon: const Icon(Icons.play_arrow),
+                      ),
+                    ),
+                    Container(
+                      decoration: BoxDecoration(
+                        shape: BoxShape.circle,
+                        color: Theme.of(context).primaryColorLight,
+                      ),
+                      child: IconButton(
```

Comment:
```suggestion
                      // Add a button to toggle navigation mode.
                      child: IconButton(
```

Reply:
I find the `needRecenter` state variable a little confusing. Could the variable instead of being a boolean, be a `LocationDisplayAutoPanMode` and then this button only be enabled if it `!= LocationDisplayAutoPanMode.Navigation` ?